packages/neon/neon/lib/src/pages/login_qrcode.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_zxing/flutter_zxing.dart';
+import 'package:neon/src/router.dart';
+import 'package:neon/src/utils/exceptions.dart';
+import 'package:neon/src/utils/login_qrcode.dart';
+import 'package:neon/src/widgets/exception.dart';
+
+class LoginQrcodePage extends StatefulWidget {
+  const LoginQrcodePage({
+    super.key,
+  });
+
+  @override
+  State<LoginQrcodePage> createState() => _LoginQrcodePageState();
+}
+
+class _LoginQrcodePageState extends State<LoginQrcodePage> {
+  String? _lastErrorURL;
+
+  @override
+  Widget build(final BuildContext context) => Scaffold(
+        appBar: AppBar(),
+        body: ReaderWidget(
```

Comment:
We should probably also specify `codeFormat: Format.qrCode` to avoid scanning wrong formats in the first place